src/Registry/TypeRegistry.php

```diff
@@ -792,6 +819,56 @@ public function register_union_type( string $type_name, array $config ): void {
 		$this->register_type( $type_name, $config );
 	}
 
+	/**
+	 * Get the keys that are prepared for introspection.
+	 *
+	 * @return array<string>
+	 */
+	protected static function get_introspection_keys(): array {
+
+		if ( null === self::$introspection_keys ) {
+			/**
+			 * Filter the keys that are prepared for introspection.
+			 *
+			 * @param array<string> $introspection_keys The keys to prepare for introspection.
+			 */
+			$introspection_keys       = \apply_filters( 'graphql_introspection_keys', [ 'description', 'deprecationReason' ] );
+			self::$introspection_keys = $introspection_keys;
+		}
+
+		return self::$introspection_keys;
+	}
+
+	/**
+	 * Prepare the config for introspection. This is used to resolve callable values for description and deprecationReason for
+	 * introspection queries.
+	 *
+	 * @param array<string,mixed> $config The config to prepare.
+	 *
+	 * @return array<string,mixed> The prepared config.
```

Comment:
Can we mark this as `@internal`?

(https://github.com/wp-graphql/wp-graphql/pull/3356#discussion_r2051293680)